test/Npgsql.Tests/PoolTests.cs

```diff
@@ -39,6 +40,97 @@ public void MinPoolSizeLargerThanMaxPoolSize()
             Assert.That(() => CreateConnection(connString), Throws.Exception.TypeOf<ArgumentException>());
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
```

Comment:
This looks like it would work better as two separate tests - substantial logic is conditioned in the test.

Reply:
I've split this into two separate tasks now